Created GopherGate Database Docker Simulation and Implement Database functionality in GopherGate Core

KShiMin opened the pull request:
## Description

Describe the purpose of this pull request and the changes made.
Provide developers with newly created GopherGate database docker compose for testing.

## Related Issue(s) and PR(s)

List any related issues or pull requests. Example: Fixes #123, Closes #456.
- Closes #46 

## Changes Made
- Renamed folder and file for dev simulation config
- Created docker compose for GopherGate database testing
- Updated .gitignore to ignore dev simulation config packets.

## Additional Notes

Provide any extra information, such as testing steps, potential impact or deployment considerations.

NIL

KShiMin committed:
chore(dev-sim/): renamed wireguard-dev to dev-sim.

This folder and file renamed is to standardise all dev environment
simulation. `.gitignore` file has also been updated to reflect the
new folder name.

KShiMin committed:
feat(dev-sim/dev-sim): Added DB config support

KShiMin commented:
## Note
- GopherGate docker compose has not been tested yet

Cyrof committed:
refactor(dev-sim/dev-sim, dev-sim/README, gitignore): Changed the
database docker logic to use docker secrets instead of `.env`. Also
updated the README for the usage guide on the new postgres database
docker

Cyrof commented:
> ## Note
> * GopherGate docker compose has not been tested yet

I have pushed up a fix for the docker compose for the database section. I also updated the `README` file on the usage guide for the database. I will continue to do up the database section in gophergate core.

Cyrof committed:
feat(gophergate-core/envx/): Created a central loader for env files
(mainly for dev)

refactor(gitignore): Ignore dev database files

Cyrof committed:
feat(gophergate-core/envx): Added a helper to load `.env` files

Cyrof committed:
feat(gophergate-core/dbx/): Created the database connections feature in
the gophergate-core to allow all other apps (agent & UI) to use the
database

refactor(gophergate-core/go.mod, gophergate-core/go.sum): Added
necessary package for dbx

Cyrof committed:
chore(gophergate-core/go.mod, gophergate-core/go.sum): Tidy up the
packages

Cyrof committed:
fix(gophergate-core/dbx): Fix wrong pgx version

Cyrof commented on dev-sim/README.md:
````diff
@@ -94,6 +94,78 @@ To stop and remove the container + network:
+Bring up the dev DB:
+
+```bash
+docker compose -f dev-sim.yaml up -d db
````
There is an extra `db` at the end of this line which is unnecessary.

KShiMin committed:
fix(dev-sim/README): Fixed docker compose command.

Cyrof reviewed (changes requested):
Since the inclusion of `envx` in gophergate-core, should use this package for logger as well.

KShiMin committed:
refactor(logx/logx): Changed mode decision logic to use envx.

Cyrof reviewed (approved):
LGTM.
